tcp_exporter.c

```diff
@@ -64,64 +169,56 @@ enum MHD_Result serve_metrics(void *cls, struct MHD_Connection *connection,
  */
 int start_http_server() {
     struct MHD_Daemon *daemon;
-    daemon = MHD_start_daemon(MHD_USE_SELECT_INTERNALLY, PORT, NULL, NULL, &serve_metrics, NULL, MHD_OPTION_END);
+    daemon = MHD_start_daemon(MHD_USE_INTERNAL_POLLING_THREAD, PORT, NULL, NULL, &serve_metrics, NULL, MHD_OPTION_END);
     if (NULL == daemon) {
         return 1;
     }
     printf("HTTP server running on port %d\n", PORT);
     return 0;
 }
 
-/**
- * @brief Updates connection counters based on the connection state.
- *
- * @param state The state of the connection (Opening, Open, Closing, Closed).
- */
-void update_connection_counters(const char *state) {
-    if (strcmp(state, "Opening") == 0) {
-        opening_connections++;
-    } else if (strcmp(state, "Open") == 0) {
-        open_connections++;
-    } else if (strcmp(state, "Closing") == 0) {
-        closing_connections++;
-    } else if (strcmp(state, "Closed") == 0) {
-        closed_connections++;
-    }
-}
-
 /**
  * @brief Logs connection events and updates counters based on state.
  *
- * @param event_type The type of event (e.g., new, update, close).
  * @param ct The connection track object.
  * @param state The state of the connection (e.g., Opening, Open, Closed).
  */
-void log_connection_event(const char *event_type, struct nf_conntrack *ct, const char *state) {
+void log_connection_event(struct nf_conntrack *ct, const char *state) {
     char src_ip[INET_ADDRSTRLEN], dst_ip[INET_ADDRSTRLEN];
-    uint32_t src, dst;
+    uint32_t src_ip_bin = nfct_get_attr_u32(ct, ATTR_ORIG_IPV4_SRC);
+    uint32_t dst_ip_bin = nfct_get_attr_u32(ct, ATTR_ORIG_IPV4_DST);
+    uint16_t src_port = ntohs(nfct_get_attr_u16(ct, ATTR_ORIG_PORT_SRC));
+    uint16_t dst_port = ntohs(nfct_get_attr_u16(ct, ATTR_ORIG_PORT_DST));
```

Comment:
TODO: ntohs also to be apply when updating metrics counter labels for prometheus exporter